### ISSUE TITLE
[Feature Request]: Find event should work with Event Subscribers

### ISSUE THREAD

**Issue by nicolasbuch** [laravel-idea/plugin#768]
### Feature Description

I've just recently discovered the awesome feature of being able to find events directly from the listener. Not only does it list them, but you are also able to click-through. This is especially great when having the same listener bound to multiple events as seen below. 
 
<img width="762" alt="image" src="https://github.com/laravel-idea/plugin/assets/19183799/3cc7ece5-58c1-4766-b2e4-63233d3321fa">

---

Sometimes it can become repetitive to bind the same listeners to multiple events directly in the `$listen` property as seen in the example below
```php
class EventServiceProvider extends ServiceProvider
{
    protected $listen = [
        ItemAddedToCart::class => [
            RemoveBundleDiscountsFromCart::class,
            ApplyBundleDiscountToCart::class,
        ],
        ItemsRemovedFromCart::class => [
            RemoveBundleDiscountsFromCart::class,
            ApplyBundleDiscountToCart::class,
        ],
        UpdatedAdForAllCartItems::class => [
            RemoveBundleDiscountsFromCart::class,
            ApplyBundleDiscountToCart::class,
        ],
    ];
}
```
In comes Event Subscribers and we can now listen to multiple events from within the same class as such

```php
class CartEventSubscriber
{
    public function handleBundle(): array
    {
        return [
            RemoveBundleDiscountsFromCart::class,
            ApplyBundleDiscountToCart::class,
        ];
    }

    public function subscribe(Dispatcher $events): array
    {
        return [
            ItemAddedToCart::class => $this->handleBundle(),
            ItemsRemovedFromCart::class => $this->handleBundle(),
            UpdatedAdForAllCartItems::class => $this->handleBundle(),
        ];
    }
}
```

But after moving to Event Subscribers the awesome feature of finding events no longer works 😢

**Comment by mgralikowski** [comment]
+1

It is not only a lack of functionality but more like a bug because it shows the icon, but after a click, there is no reaction.

![Image](https://github.com/user-attachments/assets/9931c917-b5ea-41f0-8148-4baba5044ae5)

Generally, any official construction from https://laravel.com/docs/11.x/events#writing-event-subscribers doesn't work.

**Comment by adelf** [comment]
This way is challenging to understand by code.
Laravel Idea already finds listeners if they have the `handle(ThisEvent $event)` method. However, it seems you're handling some base class of these events. Right?

**Comment by mgralikowski** [comment]
You are right, as I can assign multiple events to a single `handleSthMethod` I can't use `ThisEvent` as a parameter but rather an interface.

What is more PHPStorm/IdeaPlugin can't navigate to methods inside a subscriber.
![Image](https://github.com/user-attachments/assets/2042610b-5b0d-4637-b3d0-056373b10f37)

The only way to find listeners is to use the built-in Find Usages function.
![Image](https://github.com/user-attachments/assets/21926dee-05fb-438a-9547-da7d6ab3b33a)

**Comment by adelf** [comment]
We've implemented events fetching from subscribe and methods completion here. I hope it's enough to close this issue. The update will be released soon.

<img width="497" alt="Image" src="https://github.com/user-attachments/assets/d8fe8421-108b-430a-92ef-8cf1f70ec2fd" />

**Comment by adelf** [comment]
10.1 is released.
